### DISCUSSION TITLE
How to use NeighborSampler or any other sampler technique when each sample in minibatch is a collection of nodes (as oppose to one node)?

### DISCUSSION THREAD

**Discussion by gebrahimi91** [pyg-team/pytorch_geometric#2418] [Q&A]
Hi,

I am wondering how I can sample using NeighborSampler, when for instance the first sample in the minibatch contains nodes 1,2,3,4 and maybe the next sample contains nodes 5,6,7,8? how can i chance NeighborSampler to choose neighbors around all these nodes as opposed to just one node (sample neighbors of nodes 1,2,3,4 as opposed to just sample neighbors of node 1 (the simple case that NeighborSampler currently handles))?

Thanks so much.

**Comment by rusty1s** [comment]
The `NeighborSampler` does naturally support a set of nodes as input, as defined by `batch_size`.

**Reply by gebrahimi91** [reply]
I think i was not clear in my question. I understand what you said, but it does not solve my problem. I am saying i have a constraint on which nodes comes in some sort of grouping. for instance, if node 1 is in the batch, then i might have constraint that node 1000 and 1001 should also come in the batch if node 1 is chosen. or if node 2 in the batch, maybe nodes 101, 432, 453 also should be there. does it make sense?

**Reply by rusty1s** [reply]
Yes, that makes sense. You can either input node indices `n_id` into `NeighborSampler` that follow your ordering, and set `shuffle=False`. Otherwise, you can make directly use of `NeighborSampler.sample(n_id)`.

**Comment by wsad1** [comment]
[NeighbourSampler](https://pytorch-geometric.readthedocs.io/en/latest/modules/data.html#torch_geometric.data.NeighborSampler)  will sample neighbors around multiple nodes. For example.
```
NeighborSampler(edge_index= edge_index, sizes=[-1,-1], node_idx=torch.tensor([1,2,3,4]), batch_size=4)
```
Will get you the second hop neighborhoods for nodes 1,2,3 and 4, in the first and only batch.
